cmd/SplitGrantsGovXMLDB/handler_test.go

```diff
@@ -119,65 +114,168 @@ const SOURCE_OPPORTUNITY_TEMPLATE = `
+	<AgencyCode>TEST-AC</AgencyCode>
+	<AgencyName>Bureau of Testing</AgencyName>
+	<PostDate>09082022</PostDate>
+	<EstimatedSynopsisPostDate>02102016</EstimatedSynopsisPostDate>
+	<FiscalYear>2016</FiscalYear>
+	<EstimatedSynopsisCloseDate>04112016</EstimatedSynopsisCloseDate>
+	<EstimatedAwardDate>09082016</EstimatedAwardDate>
+	<EstimatedProjectStartDate>09302016</EstimatedProjectStartDate>
+	<LastUpdatedDate>{{.LastUpdatedDate}}</LastUpdatedDate>
+	<AwardCeiling>600000</AwardCeiling>
+	<AwardFloor>400000</AwardFloor>
+	<EstimatedTotalProgramFunding>600000</EstimatedTotalProgramFunding>
+	<ExpectedNumberOfAwards>10</ExpectedNumberOfAwards>
+	<Description>Here is a description of the opportunity.</Description>
+	<Version>Synopsis 2</Version>
+	<CostSharingOrMatchingRequirement>No</CostSharingOrMatchingRequirement>
+	<ArchiveDate>02012023</ArchiveDate>
+	<GrantorContactEmail>test@example.gov</GrantorContactEmail>
+	<GrantorContactEmailDescription>Inquiries</GrantorContactEmailDescription>
+	<GrantorContactName>Tester Person</GrantorContactName>
+	<GrantorContactPhoneNumber>800-123-4567</GrantorContactPhoneNumber>
+</OpportunityForecastDetail_1_0>
+`
+
+type grantValues struct {
+	template        string
+	OpportunityID   string
+	LastUpdatedDate string
+	isExtant        bool
+	isValid         bool
+	isForecast      bool
+}
+
+func (values grantValues) getFilename() string {
+	if values.isForecast {
+		return "v2.OpportunityForecastDetail_1_0.xml"
+	} else {
+		return "v2.OpportunitySynopsisDetail_1_0.xml"
+	}
+}
+
 func TestLambdaInvocationScenarios(t *testing.T) {
 	setupLambdaEnvForTesting(t)
 	sourceBucketName := "test-source-bucket"
 	now := time.Now()
 	s3client, cfg, err := setupS3ForTesting(t, sourceBucketName)
 	assert.NoError(t, err, "Error configuring test environment")
 
-	type grantValues struct {
-		template        string
-		OpportunityID   string
-		LastUpdatedDate string
-		isExtant        bool
-		isValid         bool
-	}
-
 	for _, tt := range []struct {
-		name        string
-		grantValues []grantValues
+		name                      string
+		isForecastedGrantsEnabled bool
+		grantValues               []grantValues
 	}{
 		{
 			"Well-formed source XML for single new grant",
+			true,
+			[]grantValues{
+				{
+					SOURCE_OPPORTUNITY_TEMPLATE,
+					"1234",
+					now.AddDate(-1, 0, 0).Format("01022006"),
+					false,
+					true,
+					false,
+				},
+			},
+		},
+		{
+			"Well-formed source XML for single new forecast",
+			true,
+			[]grantValues{
+				{
+					SOURCE_FORECAST_TEMPLATE,
+					"2345",
+					now.AddDate(-1, 0, 0).Format("01022006"),
+					false,
+					true,
+					true,
+				},
+			},
+		},
+		{
+			"When flag is disabled, ignores well-formed source XML for single new forecast",
+			false,
+			[]grantValues{
+				{
+					SOURCE_FORECAST_TEMPLATE,
+					"2345",
+					now.AddDate(-1, 0, 0).Format("01022006"),
+					false,
+					false,
+					true,
+				},
```

Comment:
This `isValid` value needs to be switched to `true` given that this `grantValues` array member is meant to generate well-formed source XML. After resolving the env var-related issue, this test case is failing `An error is expected but got nil.` due to the `require.Error()` assertion in this snippet: https://github.com/usdigitalresponse/grants-ingest/blob/971a3404d8bcbc87f5a0b1f75fc69e8dc6323808/cmd/SplitGrantsGovXMLDB/handler_test.go#L380-L390

```suggestion
				{
					SOURCE_FORECAST_TEMPLATE,
					"2345",
					now.AddDate(-1, 0, 0).Format("01022006"),
					false,
					true,
					true,
				},
```